src/ui.cpp

```diff
@@ -1327,4 +1327,79 @@ bool navigate_ui_list( const std::string &action, V &val, int page_delta, S size
         return false;
     }
     return true;
-}
\ No newline at end of file
+}
+
+template<typename V, typename S>
+inline typename std::enable_if < !std::is_enum<V>::value, V >::type increment_and_wrap( V val,
```

Comment:
I don't think moving an inline function to a source file is valid.

Reply:
Sorry that I moved them to cpp, as I was resolving the errors, it was so slow to update header files.

If that causes errors, then move them back again.
Or delete `inline` keyword, isn't it just a suggestion for the compiler?

I leave it completely to you, the reusing of function code is the only thing I wanted to address.